graph/src/data/subgraph/mod.rs

```diff
@@ -266,7 +266,7 @@ impl From<::diesel::result::Error> for SubgraphAssignmentProviderError {
 #[derive(Debug, PartialEq)]
 pub enum SubgraphAssignmentProviderEvent {
     /// A subgraph with the given manifest should start processing.
-    SubgraphStart(SubgraphManifest),
+    SubgraphStart(SubgraphName, SubgraphManifest),
```

Comment:
I'm a bit nervous about having the name back in this event. It used to be here and it ended up being used in many places, which caused problems for deduplication. Can we add some comments here to make clear that the name is only for internal debugging and may not be correct? In particular:

- Subgraph deployments can be associated with more than one name, so there may be other names
- The name is only looked up at start, so this subgraph deployment might no longer be associated with this name
- graph-node will frequently see multiple subgraph deployments with the same "name" here, because of the last bullet point (and also because of pending + current), so it can't be used as an identifier

Maybe in places where this name is used (e.g. `SubgraphAssignmentProvider::start`), it should be called something different to make clear its limitations. "logging_name"? "initial_name"?

Reply:
Alternatively the name resolution could be done closer to the elastic logger so that the risk of accidentally relying on the name for other things is reduced

Reply:
The more I think about it again, the more I think we don't want this at all. It just doesn't make sense to show the subgraph name in the logs if the deployment could be running for different subgraphs with different names at the same time.